protokube/pkg/protokube/etcd_manifest.go

```diff
@@ -30,70 +32,51 @@ func BuildEtcdManifest(c *EtcdCluster) *v1.Pod {
 	pod.Kind = "Pod"
 	pod.Name = c.PodName
 	pod.Namespace = "kube-system"
+	pod.Labels = map[string]string{"k8s-app": c.PodName}
 
-	pod.Labels = map[string]string{
-		"k8s-app": c.PodName,
-	}
-
-	// TODO another hardcoded version
-	image := "/etcd:2.2.1"
-	imageRegistry := "gcr.io/google_containers"
+	etcdImage := "/etcd:2.2.1"
+	etcdRegistry := "gcr.io/google_containers"
 
-	// Test to determine if the container registry has been passed in as a flag.
-	// If so use the provider registry location.
+	// @check if the container is being overloaded via flags
 	if c.ImageSource == "" {
-		image = imageRegistry + image
+		etcdImage = etcdRegistry + etcdImage
 	} else {
-		image = strings.TrimSuffix(c.ImageSource, "/") + image
+		etcdImage = strings.TrimSuffix(c.ImageSource, "/") + etcdImage
 	}
 
 	pod.Spec.HostNetwork = true
-
 	{
 		container := v1.Container{
 			Name:  "etcd-container",
-			Image: image,
+			Image: etcdImage,
 			Resources: v1.ResourceRequirements{
 				Requests: v1.ResourceList{
 					v1.ResourceCPU: c.CPURequest,
 				},
 			},
-			Command: []string{
-				"/bin/sh",
-				"-c",
-				"/usr/local/bin/etcd 1>>/var/log/etcd.log 2>&1",
-			},
-
-			// Note that we listen on 0.0.0.0, not 127.0.0.1, so we can support etcd clusters
-			Env: []v1.EnvVar{
-				{Name: "ETCD_NAME", Value: c.Me.Name},
-				{Name: "ETCD_DATA_DIR", Value: "/var/etcd/" + c.DataDirName},
-				{Name: "ETCD_LISTEN_PEER_URLS", Value: fmt.Sprintf("http://0.0.0.0:%d", c.PeerPort)},
-				{Name: "ETCD_LISTEN_CLIENT_URLS", Value: fmt.Sprintf("http://0.0.0.0:%d", c.ClientPort)},
-				{Name: "ETCD_ADVERTISE_CLIENT_URLS", Value: fmt.Sprintf("http://%s:%d", c.Me.InternalName, c.ClientPort)},
-				{Name: "ETCD_INITIAL_ADVERTISE_PEER_URLS", Value: fmt.Sprintf("http://%s:%d", c.Me.InternalName, c.PeerPort)},
-				{Name: "ETCD_INITIAL_CLUSTER_STATE", Value: "new"},
-				{Name: "ETCD_INITIAL_CLUSTER_TOKEN", Value: c.ClusterToken},
-			},
+			Command: []string{"/usr/local/bin/etcd"},
 		}
-
-		var initialCluster []string
-		for _, node := range c.Nodes {
-			// TODO: Use localhost for ourselves?  Does the cluster view have to be symmetric?
-			initialCluster = append(initialCluster, node.Name+"="+fmt.Sprintf("http://%s:%d", node.InternalName, c.PeerPort))
-		}
-		container.Env = append(container.Env, v1.EnvVar{Name: "ETCD_INITIAL_CLUSTER", Value: strings.Join(initialCluster, ",")})
+		// build the the environment variables for etcd service
+		container.Env = buildEtcdEnvironmentOptions(c)
 
 		container.LivenessProbe = &v1.Probe{
 			InitialDelaySeconds: 15,
 			TimeoutSeconds:      15,
 		}
-		container.LivenessProbe.HTTPGet = &v1.HTTPGetAction{
-			Host: "127.0.0.1",
-			Port: intstr.FromInt(c.ClientPort),
-			Path: "/health",
+		// ensure we have the correct probe schema
+		if c.isTLS() {
+			container.LivenessProbe.TCPSocket = &v1.TCPSocketAction{
+				Host: "127.0.0.1",
```

Comment:
Makes sense.  I checked on how the GCE configuration does it, and AFAICT they don't use TLS for clients (which are always over localhost), just for peer-to-peer, and then they health-check on the client-port.  Your approach is better IMO.